test/integration/apiserver/apply/apply_crd_test.go

```diff
@@ -0,0 +1,544 @@
+							"type": "object",
+							"properties": {
+								"cronSpec": {
+									"type": "string",
+									"pattern": "^(\\d+|\\*)(/\\d+)?(\\s+(\\d+|\\*)(/\\d+)?){4}$"
+								}
+							}
+						}, {
+							"type": "string"
+						}
+					]
+				}
+			}
+		}
+	}`), &c)
+	if err != nil {
+		t.Fatal(err)
+	}
+	noxuDefinition.Spec.Validation = &c
+
+	noxuDefinition, err = fixtures.CreateNewCustomResourceDefinition(noxuDefinition, apiExtensionClient, dynamicClient)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	kind := noxuDefinition.Spec.Names.Kind
+	apiVersion := noxuDefinition.Spec.Group + "/" + noxuDefinition.Spec.Version
+	name := "mytest"
+
+	rest := apiExtensionClient.Discovery().RESTClient()
+	yamlBody := []byte(fmt.Sprintf(`
+apiVersion: %s
+kind: %s
+metadata:
+  name: %s
+  finalizers:
+  - test-finalizer
+spec:
+  cronSpec: "* * * * */5"
+  replicas: 1`, apiVersion, kind, name))
+	result, err := rest.Patch(types.ApplyPatchType).
+		AbsPath("/apis", noxuDefinition.Spec.Group, noxuDefinition.Spec.Version, noxuDefinition.Spec.Names.Plural).
+		Name(name).
+		Param("fieldManager", "apply_test").
+		Body(yamlBody).
+		DoRaw()
+	if err != nil {
+		t.Fatalf("failed to create custom resource with apply: %v:\n%v", err, string(result))
+	}
+
+	// Patch object to add another finalizer to the finalizers list
+	result, err = rest.Patch(types.MergePatchType).
+		AbsPath("/apis", noxuDefinition.Spec.Group, noxuDefinition.Spec.Version, noxuDefinition.Spec.Names.Plural).
+		Name(name).
+		Body([]byte(`{"metadata":{"finalizers":["another-one"]}}`)).
+		DoRaw()
+	if err != nil {
+		t.Fatalf("failed to add finalizer with merge patch: %v:\n%v", err, string(result))
+	}
+
+	// Re-apply the same config, should work fine, since finalizers should have the list-type extension 'set'.
+	result, err = rest.Patch(types.ApplyPatchType).
+		AbsPath("/apis", noxuDefinition.Spec.Group, noxuDefinition.Spec.Version, noxuDefinition.Spec.Names.Plural).
+		Name(name).
+		Param("fieldManager", "apply_test").
+		SetHeader("Accept", "application/json").
+		Body(yamlBody).
+		DoRaw()
+	if err != nil {
+		t.Fatalf("failed to apply same config after adding a finalizer: %v:\n%v", err, string(result))
+	}
+
+	// Test that the finalizers were not overwritten
+	obj := unstructured.Unstructured{}
+	err = obj.UnmarshalJSON(result)
+	if err != nil {
+		t.Fatalf("failed to unmarshal response: %v", err)
+	}
+	if actual, expected := len(obj.GetFinalizers()), 2; actual != expected {
+		t.Fatalf("expected %v finalizers after re-applying but got %v:\n%v", expected, actual, string(result))
+	}
+
+	// Patch object to change the number of replicas
+	result, err = rest.Patch(types.MergePatchType).
+		AbsPath("/apis", noxuDefinition.Spec.Group, noxuDefinition.Spec.Version, noxuDefinition.Spec.Names.Plural).
+		Name(name).
+		Body([]byte(`{"spec":{"replicas": 5}}`)).
+		DoRaw()
+	if err != nil {
+		t.Fatalf("failed to update number of replicas with merge patch: %v:\n%v", err, string(result))
+	}
```

Comment:
check spec.replica's value, to prove that the test condition is set up as expected.